### PR TITLE
Add analytics to settings options

### PR DESCRIPTION
**Pull request by kenzieschmoll**
This also refactors the settings options into stateless widgets

### CODE REVIEW

File: packages/devtools_app/lib/src/app.dart [hunk 1]
```diff
@@ -542,26 +542,34 @@ class SettingsDialog extends StatelessWidget {
         mainAxisSize: MainAxisSize.min,
         crossAxisAlignment: CrossAxisAlignment.start,
         children: [
-          _buildOption(
+          CheckboxSetting(
             label: const Text('Use a dark theme'),
             listenable: preferences.darkModeTheme,
             toggle: preferences.toggleDarkModeTheme,
+            gaEnabledTag: analytics_constants.darkThemeEnable,
```
Comment (jacob314) [inline packages/devtools_app/lib/src/app.dart:549]: seems verbose adding a separate tag for enable and disable. Can we just pass in a prefix.

Comment (kenzieschmoll) [reply packages/devtools_app/lib/src/app.dart]: done

**Review by jacob314 — APPROVED** [review]
lgtm

**Comment by kenzieschmoll** [comment]
Work towards #3270

**Pull request merged**